src/Language/Docker/Parser/Run.hs

```diff
@@ -0,0 +1,288 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TupleSections #-}
+
+module Language.Docker.Parser.Run
+  ( parseRun,
+    runFlags,
+  )
+where
+
+import Data.Functor (($>))
+import qualified Data.Set as Set
+import Language.Docker.Parser.Arguments (arguments)
+import Language.Docker.Parser.Prelude
+import Language.Docker.Syntax
+
+data RunFlag
+  = RunFlagMount RunMount
+  | RunFlagSecurity RunSecurity
+  | RunFlagNetwork RunNetwork
+  deriving (Show)
+
+data RunMountArg
+  = MountArgFromImage Text
+  | MountArgGid Integer
+  | MountArgId Text
+  | MountArgMode Text
+  | MountArgReadOnly Bool
+  | MountArgRequired
+  | MountArgSharing CacheSharing
+  | MountArgSource SourcePath
+  | MountArgTarget TargetPath
+  | MountArgType Text
+  | MountArgUid Integer
+  deriving (Show)
+
+data MountType
+  = Bind
+  | Cache
+  | Tmpfs
+  | Secret
+  | Ssh
+
+parseRun :: Parser (Instruction Text)
+parseRun = do
+  reserved "RUN"
+  Run <$> runArguments
+
+runArguments :: Parser (RunArgs Text)
+runArguments = do
+  presentFlags <- choice [runFlags <* requiredWhitespace, pure (RunFlags Nothing Nothing Nothing)]
+  args <- arguments
+  return $ RunArgs args presentFlags
+
+runFlags :: Parser RunFlags
```

Comment:
Does this mean that we will support experimental `buildkit` syntax?  I have found it only here https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/experimental.md. Do you have some better resource?

Reply:
Yes we do support them now!

I used that file, but as usual with the dockerfile syntax, I ended up reading the source code to discover lots more.

Reply:
cool, thanks